### PR TITLE
Empty trash feature

### PR DESCRIPTION
**Pull request by goldoraf**
This PR adds the ability to empty the trash folder.

#### Checklist

Before merging this PR, the following things must have been done:

- [ ] Faithful integration of the mockups at all screen sizes
- [ ] Tested on supported browsers, including responsive mode (IE11 and Edge, Chrome >= 42, 2 latest versions of Firefox and Safari)
- [ ] Localized in english and french
- [ ] All changes have test coverage

### CODE REVIEW

File: src/ducks/trash/Toolbar.jsx [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
+import EmptyTrashConfirm from '../../components/EmptyTrashConfirm'
+import Spinner from '../../components/Spinner'
 
-const Toolbar = ({ t, disabled, onSelectItemsClick }) => (
+import { emptyTrash } from './actions'
+
+class DeleteButton extends Component {
+  state = {
+    working: false
+  };
```
Comment (kosssi) [inline src/ducks/trash/Toolbar.jsx]: I thinks your have an extra ';' and can you skip a line after?

File: src/actions/index.js [hunk 2]
```diff
@@ -60,13 +60,14 @@ export const openFolder = (folderId) => {
       const folder = await cozy.client.files.statById(folderId, offline)
       const parentId = folder.attributes.dir_id
       const parent = !!parentId && await cozy.client.files.statById(parentId, offline)
+      // folder.relations('contents') returns null when the trash is empty
```
Comment (kosssi) [inline src/actions/index.js]: can you remove this comment?

File: src/components/SelectionBar.jsx [hunk 3]
```diff
@@ -21,7 +22,7 @@ const SelectionBar = ({ t, selected, actions, onClose, onActionComplete, onMoreC
           onClick={() => {
             actions[actionName]()
               .then(() => onActionComplete())
-              .catch(err => { console.log(err) })
+              .catch(() => onActionAbort())
```
Comment (kosssi) [inline src/components/SelectionBar.jsx]: you do nothing?

File: src/ducks/trash/actions.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/* global cozy */
+
+export const EMPTY_TRASH = 'EMPTY_TRASH'
+export const EMPTY_TRASH_SUCCESS = 'EMPTY_TRASH_SUCCESS'
+export const EMPTY_TRASH_FAILURE = 'EMPTY_TRASH_FAILURE'
+
+export const emptyTrash = () => {
```
Comment (kosssi) [inline src/ducks/trash/actions.js]: can you refactor to `export const emptyTrash = () => async dispatch => {`?

File: src/ducks/trash/Toolbar.jsx (hunk 1 above)
Comment (goldoraf) [inline src/ducks/trash/Toolbar.jsx]: The extra ';' is required by the standard for class properties, this sucks but it's necessary

File: src/components/SelectionBar.jsx (hunk 3 above)
Comment (goldoraf) [inline src/components/SelectionBar.jsx]: Yep, the catch here is required for linting reasons only...

File: src/ducks/trash/actions.js (hunk 4 above)
Comment (goldoraf) [inline src/ducks/trash/actions.js]: Yep

File: src/components/Empty.jsx [hunk 5]
```diff
@@ -3,13 +3,24 @@ import styles from '../styles/empty'
...
     <div class={styles['fil-empty']}>
       <h2>{ t('empty.title') }</h2>
       {canUpload && <p>{ t('empty.text')}</p>}
     </div>
   )
-}
+})
 
-export default translate()(Empty)
+export default Empty
```
Comment (kosssi) [inline src/components/Empty.jsx]: why you didn't write export default translate()(Empty)?

File: src/components/Empty.jsx [hunk 6]
```diff
@@ -3,13 +3,24 @@ import styles from '../styles/empty'
...
+const EmptyTrash = translate()(({ t }) => {
+  return (
+    <div class={styles['fil-trash-empty']}>
+      <h2>{ t('empty.trash.title') }</h2>
+      <p>{ t('empty.trash.text')}</p>
+    </div>
+  )
+})
+
+export { EmptyTrash }
```
Comment (kosssi) [inline src/components/Empty.jsx]: same comment?

**Pull request merged**